### DISCUSSION TITLE
Add version dropdown to allow switching between different doc versions

### DISCUSSION THREAD

**Discussion by Nadine2016** [colinwilson/lotusdocs#57] [Feature Requests]
Because we write docs for a product that goes through regular release cycles, we need to have a way to display different versions of the docs to the user. We would love to have a way to add a version drop down to the site that allows users to switch between versions. 

Docsy uses the params section in the config.toml file to configure the URLs for the different versions. https://www.docsy.dev/docs/adding-content/versioning/.

**Comment by colinwilson** [comment]
I'll definitively add this to the roadmap. Very useful and seems fairly easy to implement.

**Reply by eaguad1337** [reply]
Hello @colinwilson! I'd really like to have this too, are you working on it? I ask because I could check it out and maybe contribute to add this feature.

**Reply by colinwilson** [reply]
> are you working on it?

Yep. Got a big update coming soon that'll address this, other feature requests here, the docs, and PR's.

**Reply by eaguad1337** [reply]
I can't wait to see it!

**Reply by Nadine2016** [reply]
so exciting. Thank you for working on this!

**Reply by Nadine2016** [reply]
@colinwilson just wanted to check if you already have an idea when this version switcher could become available?

**Comment by life1347** [comment]
@Nadine2016 I spent some time this weekend to add this features. Please check the PR https://github.com/colinwilson/lotusdocs/pull/150

**Comment by Nadine2016** [comment]
@life1347 thank you so much for working on this. I tried out the change. the drop down looks great. however, the linking does not seem to be working correctly for me. I have the following set in the toml file 

```
[params.versionInfo]
  dropdownMenuText = "Releases"
  archived = false                     # indicate this is an archived doc set branch and display warning banner - default false
  unreleased = true                   # indicate this is an unreleased doc set branch and display warning banner - default false

[[params.versions]]
  version = "main"
  url = "https://mydocs.com/main/"

[[params.versions]]
  version = "v2.5.x"
  current = true                       # display as current branch version

[[params.versions]]
  version = "v2.5.x"
  url = "https://mydocs.com/main/2.5.x/"
```

it looks like this for me: 
<img width="295" alt="image" src="https://github.com/colinwilson/lotusdocs/assets/17709352/e4d70831-5415-47d1-a763-d7953f435105">


however, when I select main for example, I am getting redirected to `https://mydocs.com/main/2.5.x/about/architecture/` 
<img width="784" alt="image" src="https://github.com/colinwilson/lotusdocs/assets/17709352/71c46ea0-d861-4852-8754-a8e9b126cc63">

I am also not sure where the about/architecture piece is coming from. it is not part of my hugo toml file. 

I will go through everything again tomorrow, but maybe you already know what might be causing this.

**Reply by Nadine2016** [reply]
ideally, I would like to add the current page I am on to the url I set in the params.version section. so let's say if I am on `https://mydocs.com/main/about/architecture/` and I switch to version 2.5.x, I would expect the URL to look like this `https://mydocs.com/2.5.x/about/architecture/` > basically taking the version URL from the toml file and appending the page that I am on right now. 

Not sure if this is what is intended here and that's why the about/architecture page not matter where I am in the doc set.
